### ISSUE TITLE
Rating methods comparison

### ISSUE THREAD

**Issue by amirrr** [Watts-Lab/commonsense-statements#33]
![image](https://github.com/user-attachments/assets/bcdc2387-eb0a-4aca-a4c4-a07d755a8bac)

**Comment by amirrr** [comment]
| gpt vs llama3 | llama3 vs llama2 | gpt vs llama2 |
|---------|---------|---------|
| ![Image 1](https://github.com/user-attachments/assets/eb00472b-d871-43a4-ba14-518dd66212c4) | ![Image 2](https://github.com/user-attachments/assets/e4a0fc64-80f6-492e-8c89-2897e0b1958d) | ![Image 3](https://github.com/user-attachments/assets/0b1dc3c0-d51e-474e-81bf-1ba89a83c2f0) |

**Comment by joshnguyen99** [comment]
![SCR-20241203-gjq](https://github.com/user-attachments/assets/e218bae8-d598-4b9a-adf0-a3ec63c10d58)

@amirrr @markwhiting Here's a comparison between humans and LLaMA-3.1 in labeling 6 dimensions for statements in our previous corpus.

Note that `figure_of_speech` is the negative class for `literal language`. If it were the positive class, the F-1 score would just be 0.48.

**Comment by markwhiting** [comment]
Thanks, do these look right to you? I note that we did have reversed classes for some features that we reverse in the start of the R script from the prior experiment, and just want to ensure that the ratings you're comparing with reflect the correct state.

**Comment by joshnguyen99** [comment]
I think they should be correct. In the original data file (and R script), only `everyday` was coded as 1. Every other variable should be flipped.

**Comment by markwhiting** [comment]
OK, thanks. Qualitatively do you have any qualms with this? 

Also, how much will this change the design points we can cover with our current corpus?

**Comment by joshnguyen99** [comment]
The numbers are lower than I expected, although upon manual inspection some human-given labels may not be reasonable.

For example:
- "We should ensure police have body cameras on them." -> Humans said `fact`, LLM said `opinion`
- "The first thing you do when you have an asthma attack is use the inhaler." -> Humans said `social`, LLM said `physical`

I'm not sure how much the design points will have to change, but I think we should do a manual evaluation of LLaMA-3.1's labels in the new dataset. Perhaps we can sample 200 statements, label them ourselves independently (at least by me and Amir), and then check whether we agree with each other and with the LLM.

**Comment by markwhiting** [comment]
Thanks — yeah a manual pass seems worth it. At worst we will then have some ground truth to work with.

If we want to scale that a bit more we can also put a more sophisticated version of the task in front of turkers.

**Comment by joshnguyen99** [comment]
TODO:
- Dataset for statements is [here](https://github.com/Watts-Lab/commonsense-data/blob/main/statements/statements_1.csv).
- Select only statements for which `published` is True. These are statements that are on the platform already.
- Randomly select 200 statements. Try multiple ways.
- Coordinate labeling.
- Prompt different LLMs for classification.

**Comment by joshnguyen99** [comment]
@markwhiting @amirrr —

I have prompted LLaMA-3-8B to classify all N = 10,110 statements [here](https://github.com/Watts-Lab/commonsense-data/blob/main/statements/statements_1.csv).

The same features previously obtained by Amir are in [this file](https://github.com/Watts-Lab/commonsense-data/blob/main/statementproperties/statementproperties_1.csv). Note that they are only available for the first N = 8,814 statements.

For comparison, I have put them together. For feature `X`, `X_old` is Amir's version, while `X` is mine. Note that `X_old` is only available for the first 8,814 rows.

[statements_features_comparison.csv](https://github.com/user-attachments/files/18215580/statements_features_comparison.csv)

**Comment by joshnguyen99** [comment]
Here are 200 statements randomly sampled from those for which `published` is True. I will label them myself and hope to get @amirrr's ratings soon as well.

[to_label_200.csv](https://github.com/user-attachments/files/18215693/to_label_200.csv)

**Comment by amirrr** [comment]
> Here are 200 statements randomly sampled from those for which `published` is True. I will label them myself and hope to get @amirrr's ratings soon as well.
> 
> [to_label_200.csv](https://github.com/user-attachments/files/18215693/to_label_200.csv)

Will start working on them now.

**Comment by joshnguyen99** [comment]
@markwhiting @amirrr 

This is for the 200-statement dataset that Amir and I annotated independently.

I used LLaMA-3.1-8B to make predictions for the 6 dimensions. The script can be found [here](https://github.com/Watts-Lab/commonsense-lm/blob/main/dimensions-prediction-lm/feature_prompts_inidividual.py).

Essentially I asked LLaMA to output either of the two tokens: `0` and `1`. Then I performed a softmax transformation on the logit scores of only these two tokens to get the associated probabilities.

The figure attached here shows the ROC curves for these scores. There are two settings: when using my labels as the ground truth (black), and when using Amir's labels (blue).

![Image](https://github.com/user-attachments/assets/ed60ab87-6368-4153-9cac-56ba3a661355)

**Comment by joshnguyen99** [comment]
I performed the same analysis for the original 4,407-statement dataset. Here I treat the **mode** of human annotations as the ground truth, similar to what we've done so far.

![Image](https://github.com/user-attachments/assets/99ad8023-4466-46af-9e04-92aeaf17396f)

I also use the **mean** of these annotations. For example, if a statement gets 5 annotations, 3 of which say "physical" and 2 say "social", then the human score for "physical" is 3 / 5 = 60%. The figure below shows the correlation between LLaMA scores and human scores. I use the Spearman correlation here, and all of them are statistically significant.

![Image](https://github.com/user-attachments/assets/b7614c38-19c9-4a7f-93dc-23b6c24d8b20)

**Comment by markwhiting** [comment]
Interesting. Thanks for this. It feels like we still don't have something that is strongly aligned, but I wonder what the best next step is for getting to a place where we are satisfied.

**Comment by markwhiting** [comment]
Would you mind doing the average analysis on the 200 that you two rated?

**Comment by joshnguyen99** [comment]
Sure. Since there are only two annotators, the "human scores" are too coarse to be treated as continuous. Therefore, I determine a statement to be positive if at least Amir or I think it's positive, effectively doing an OR operation.

The ROC curves are as follows.

![Image](https://github.com/user-attachments/assets/82b77434-4229-4fe4-8a40-611f43d0ec05)

**Comment by joshnguyen99** [comment]
I would say the following categories are pretty robust: `fact`, `physical` and `literal language`. Perhaps `everyday` as well.

The hardest for me to annotate were:
- `positive`/`normative`: because some statements do not contain directive words like "should", "must" and "ought to", but could the normativity of them could still be inferred.

-`knowledge`/`reasoning`: similarly, should we only be relying on connectives like `therefore`, `thus` or `hence`?

**Comment by markwhiting** [comment]
I think even the resolution of three possible values 0, .5 or 1, I'm hoping that we see improved alignment over the binary case, because I think that would let us argue that the continuous categories from the model are sufficiently more informative that it's worth treating those as our standard. 

So I'd still be interested to look at that if you can easily run it. 

Another option would be to take these 200 statements and a few rounds of human raters on them to see if we get more consistent responses that way too. 

Lastly, I really don't mind if we want to try to improve our definitions of these concepts as I think that will benefit us as well as the models.

**Comment by joshnguyen99** [comment]
Sure, here it is!

![Image](https://github.com/user-attachments/assets/66d8f9ac-df64-4d91-a54e-a61a8e525a85)

I agree with you. I think we can all improve on our ratings in the next iteration, with a different set of statements. I certainly would.

**Comment by markwhiting** [comment]
Thank you. 

Interesting that the model sometimes captures disagreement nicely and sometimes does not. Also interesting that LLaMA's probability for some is skewed so high. I wonder if we should post tune those to have better coverage and if that would help at all.

`positive` seems like the hardest one here because neither direction of tuning will help us. 

Is this after a revisit to the ratings (I think @amirrr was planning to look at `positive` again?)
